packages/ERTP/test/unitTests/test-issuerObj.js

```diff
@@ -183,126 +171,112 @@ test('issuer.burn', t => {
   const { issuer, mint, amountMath } = makeIssuerKit('fungible');
   const payment1 = mint.mintPayment(amountMath.make(837));
 
-  E(issuer)
+  return E(issuer)
     .burn(payment1, amountMath.make(837))
     .then(burntBalance => {
-      t.ok(
+      t.assert(
         amountMath.isEqual(burntBalance, amountMath.make(837)),
         `entire minted payment was burnt`,
       );
-      t.rejects(() => issuer.getAmountOf(payment1), /payment not found/);
+      return t.throwsAsync(() => issuer.getAmountOf(payment1), { message: /payment not found/ });
     })
     .catch(e => t.assert(false, e));
 });
```

Comment:
t.notThrows() would be good here.

Reply:
Would our rule be to put `t.throwsAsync` or `t.notThrows` (`t.notThrowsAsync`??) around every call that returns a Promise? I can't imagine us being ok with any potential rejection going unnoticed, so every promise would fall into one of:

* we expect a positive result and want to examine it: use `checkResult(await ..)` or `t.notThrows(..).then(checkResult)`
* we expect a positive result but don't care about the value: use `await ..` or `t.notThrows(..);`
* we expect a negative result (testing delibrary failure situations): use `t.throwsAsync(..);`

For sync stuff, I expect an exception that throws high enough to be noticed by the enclosing test case to flunk the test. As long as any lower-level exceptions cause a Promise to reject, and we don't drop any promises on the floor inside our libraries, and our test cases are on the lookout for rejections, then arbitrary exceptions in our libraries ought to make the test cases flunk too.